const cart = ["jeans"];

createOrder(
  cart,
  (orderId) => {
    console.log("order created Succesfully");
    proccedToPayment(orderId);
  },
  (error) => {
    console.log("validation failed", error.message);
  }
);

function validateCart(cart) {
  return false;
  // return true
}

function createOrder(cart, successCallBack, errorCallback) {
  if (!validateCart(cart)) {
    const err = new Error("Cart not valid");
    errorCallback(err); // Call error callback
    return;
  }
  const orderId = "12345"; // Normally generated by DB
  successCallBack(orderId);
}
